### ISSUE TITLE
UMat::UMat(commaInitializer) is declared but has no definition/implementation

### ISSUE THREAD

**Issue by diablodale** [opencv/opencv#18879]
Hi all. The codebase declares but does not define/implement the UMat constructor for a `MatCommaInitializer_` comma initializer.

https://github.com/opencv/opencv/blob/049b50d9c0dfb5d06d7528ea334cf42c7bc435f6/modules/core/include/opencv2/core/mat.hpp#L2412

that has no definition. git blame shows that declaration was added 7 years ago, but I can find no matching definition.
Therefore when code using that constructor is compiled, it results in errors like
```
error LNK2001: unresolved external symbol "public: __cdecl cv::UMat::UMat<float>(class cv::MatCommaInitializer_<float> const &)" (??$?0M@UMat@cv@@QEAA@AEBV?$MatCommaInitializer_@M@1@@Z)
fatal error LNK1120: 1 unresolved externals
```

##### System information (version)
- OpenCV => 4.5.0
- Operating System / Platform => Windows 10 64-Bit
- Compiler => Visual Studio Community 2019 v16.8.2

##### Steps to reproduce

Create a UMat with a comma initializer like the following...

```
cv::UMat mymatrix {(cv::Mat_<float>(3, 3) <<
                    1.f, 0.f, 0.f,
                    0.f, 1.f, 0.f,
                    0.f, 0.f, 1.f)};
```

##### Issue submission checklist

 - [X] I report the issue, it's not a question
 - [X] I checked the problem with documentation, FAQ, open issues,
       answers.opencv.org, Stack Overflow, etc and have not found solution
 - [X] I updated to latest OpenCV version and the issue is still there
 - [X] There is reproducer code and related data files: videos, images, onnx, etc

**Comment by alalek** [comment]
Thank you for the report!

This definition should be removed. Let keep UMat interface simple.

**Comment by diablodale** [comment]
I can make a PR to remove that definition. I've got a testing harness for OpenCV setup now.
Should I make it against master and you will cherry pick it back to 3.4? Or?

**Comment by alalek** [comment]
Great!
We have regular merge of bug fixes from `3.4` into `master`. So `3.4` target branch should work (code should be C++98 compatible on this branch).
`master` branch should work too. We will backport (cherry pick) necessary changes after merge.
